src/services/createRouter.ts

```diff
@@ -150,6 +145,33 @@ export function createRouter<
         throw new Error(`Switch is not exhaustive for before hook response status: ${JSON.stringify(beforeResponse satisfies never)}`)
     }
 
+    if (!isExternal(url)) {
+      setPropsAndUpdateRoute(navigationId, to, from)
+    }
+
+    const afterResponse = await hooks.runAfterRouteHooks({ to, from })
+
```

Comment:
This seems like trying to support after hooks when navigating to external routes. Does that work? I expect that as soon as history.update is called the browser starts navigating away.

Reply:
added a test to confirm this does in fact work, take a look at `hooks.browser.spec.ts` "external route hooks are called correctly" in last commit